packages/compile-common/src/index.ts

```diff
@@ -1,2 +1,62 @@
 import { Profiler } from "./profiler";
 export { Profiler };
+
+type Compilation = {
+  sourceIndexes: string[];
+  contracts: CompiledContract[];
+  compiler: {
+    name: string;
+    version: string;
+  };
+};
+
+type CompilerResult = Compilation[];
+
+type CompiledContract = {
+  contractName: string;
+  contract_name: string;
+  sourcePath: string;
+  source: string;
+  sourceMap: string;
+  deployedSourceMap: string;
+  legacyAST: object;
+  ast: object;
+  abi: object[];
+  metadata: string;
+  bytecode: string;
+  deployedBytecode: string;
```

Comment:
These should be the new bytecode format, which looks like this:

```typescript
interface Bytecode {
  bytes: string;
  linkReferences: LinkReference[];
}

interface LinkReference {
  offsets: number[];
  name: string | null; // this will be the contractName of the library or some other identifier
  length: number;
}
```

Reply:
Ok, I'm going to lift this code snippet.